### PR TITLE
Feature/ci unique openwrt builder names

### PR DESCRIPTION
**Pull request by rmelotte**
- Make it possible to choose the image tag. While doing that, switch to using command line options instead of files when calling the build script.
- Use unique names for the openwrt builder, so that we can safely run multiple jobs at the same time. 

This should make the CI pipelines to be picked up faster, since it will no longer be limited to 1 job at a time (once we make the related change to the runner).

### CODE REVIEW

File: tools/docker/builder/openwrt/build.sh
Comment (tomereli) [inline tools/docker/builder/openwrt/build.sh]: short options are missing some of the long options

File: tools/docker/builder/openwrt/build.sh
Comment (tomereli) [inline tools/docker/builder/openwrt/build.sh:97]: Why do we need empty exports?

File: tools/docker/builder/openwrt/build.sh
Comment (tomereli) [inline tools/docker/builder/openwrt/build.sh]: I prefer to declare all defaults outside of `main()` where we put `VERBOSE`, then simply update them directly in the `getopt` loop if selected.

File: tools/docker/builder/openwrt/build.sh
Comment (tomereli) [inline tools/docker/builder/openwrt/build.sh]: I prefer to add this `case` to the `while` loop above and treat it like any other argument - by overriding defaults which are declared outside of `main`.

File: tools/docker/builder/openwrt/build.sh
Comment (tomereli) [inline tools/docker/builder/openwrt/build.sh:32]: I think it is better to explicitly pass parameters to this function and use $1 $2 etc. into local parameters (nitpick)

File: .gitlab-ci.yml
Comment (tomereli) [inline .gitlab-ci.yml]: I think its better to change `$TARGET` to `$TARGET_DEVICE` since it will be more readable this way IMO.

**Review by tomereli — APPROVED** [review]
Mainly nitpicks so already approving, please fix the comments or reply before you merge.

File: tools/docker/builder/openwrt/build.sh
Comment (rmelotte) [reply tools/docker/builder/openwrt/build.sh:97]: They need to be exported for `docker build` to be able to use them. The other option is to use `docker build --build-arg OPTION=$OPTION` everywhere (i.e. specifying the arg value explicitly), but I didn't want to type the args twice everywhere.
If there is a good reason to do it I can still change it.

File: tools/docker/builder/openwrt/build.sh
Comment (rmelotte) [reply tools/docker/builder/openwrt/build.sh]: Done except for "image_tag", which depend on other options.

File: tools/docker/builder/openwrt/build.sh
Comment (rmelotte) [reply tools/docker/builder/openwrt/build.sh]: Done.

File: tools/docker/builder/openwrt/build.sh
Comment (rmelotte) [reply tools/docker/builder/openwrt/build.sh]: Done.

File: .gitlab-ci.yml
Comment (rmelotte) [reply .gitlab-ci.yml]: Done in a separate commit, since the variables are already in master, and the changes in this PR are not related.

File: tools/docker/builder/openwrt/build.sh
Comment (tomereli) [inline tools/docker/builder/openwrt/build.sh]: missing a space..

File: tools/docker/builder/openwrt/build.sh
Comment (tomereli) [inline tools/docker/builder/openwrt/build.sh:118]: Great!

**Review by tomereli — APPROVED** [review]
LGTM

File: tools/docker/builder/openwrt/build.sh
Comment (rmelotte) [reply tools/docker/builder/openwrt/build.sh]: Crap, tabs vs spaces again. Let me untabify it.

**Comment by rmelotte** [comment]
Don't merge yet, the last commit is missing something (the fixup also needs to be squashed ofc).

**Comment by rmelotte** [comment]
Rebased on master, squashed the fixups.
There are 2 open review comments, other than I think it can be merged.

**Pull request merged**
